Partitioner.hpp

```diff
@@ -163,9 +168,73 @@ class LIB_EXPORT Partitioner {
     // Vector of maps of neighbours to their halo sizes after partitioning
     std::vector<std::map<int, int>> _neighbours = std::vector<std::map<int, int>>(NNBRS);
 
+    // Vector of maps of neighbours to their halo start indices after partitioning
+    std::vector<std::map<int, int>> _halo_starts = std::vector<std::map<int, int>>(NNBRS);
+
     // Vector of maps of periodic neighbours to their halo sizes after partitioning
     std::vector<std::map<int, int>> _neighbours_p = std::vector<std::map<int, int>>(NNBRS);
 
+    // Vector of maps of periodic neighbours to their halo start indices after partitioning
+    std::vector<std::map<int, int>> _halo_starts_p = std::vector<std::map<int, int>>(NNBRS);
+
+private:
+    /*!
+     * @brief Check if two domains are neighbouring. If true, then domain 2 is the [edge] neighbour
+     * of domain 1, relative to domain 1. e.g., if domain 2 is to the right of domain 1, the
+     * following call will return true:
+     * is_neighbour(d1, d2, RIGHT)
+     *
+     * @param d1 first domain
+     * @param d2 second domain
+     * @param edge LEFT, RIGHT, BOTTOM or TOP
+     * @param is_px are we looking for periodic neighbour in x-direction?
+     * @param is_py are we looking for periodic neighbour in y-direction?
+     * @return bool
+     */
+    bool is_neighbour(const Domain d1, const Domain d2, const Edge edge, const bool is_px = false,
+        const bool is_py = false);
+
+    /*!
+     * @brief Compute the start location of the halo for a given pair of neighbouring domains.
+     *
+     * For example, in the diagram above. If we want to compute the start location for the halo of
```

Comment:
There is no diagram above?

Reply:
done [3927154](https://github.com/nextsimhub/domain_decomp/pull/67/commits/392715476e16ae1d096923efbd1005d833b5e347)